axelor-stock/src/main/java/com/axelor/apps/stock/service/StoredProductsServiceImpl.java

```diff
@@ -0,0 +1,224 @@
+      BigDecimal qty = storedProducts.getStoredQty() != null ? storedProducts.getStoredQty() : null;
+      Unit unit = storedProducts.getUnit() != null ? storedProducts.getUnit() : null;
+      StockMove stockMove =
+          stockMoveService.createStockMove(
+              fromAddress,
+              toAddress,
+              company,
+              fromStockLocation,
+              toStockLocation,
+              LocalDate.now(),
+              LocalDate.now(),
+              null,
+              StockMoveRepository.TYPE_INTERNAL);
+      StockMoveLine stockMoveLine =
+          stockMoveLineService.createStockMoveLine(
+              stockMove,
+              product,
+              trackingNumber,
+              qty,
+              qty,
+              unit,
+              null,
+              fromStockLocation,
+              toStockLocation);
+      stockMove.setMassStockMove(storedProducts.getMassStockMove());
+      stockMoveService.plan(stockMove);
+      stockMoveService.realize(stockMove);
+      stockMoveRepository.save(stockMove);
+      stockMoveLineRepository.save(stockMoveLine);
+      storedProducts.setStockMoveLine(stockMoveLine);
+      storeProductsRepository.save(storedProducts);
+    } else {
+      if (storedProducts.getStoredQty().compareTo(BigDecimal.ZERO) == 0
+          && storedProducts.getTrackingNumber() != null) {
+        throw new AxelorException(
+            0,
+            I18n.get(StockExceptionMessage.STORED_QUANTITY_IS_ZERO)
+                + " "
+                + storedProducts.getStoredProduct().getFullName()
+                + " "
+                + storedProducts.getTrackingNumber().getTrackingNumberSeq());
+      } else if (storedProducts.getStoredQty().compareTo(BigDecimal.ZERO) == 0
+          && storedProducts.getTrackingNumber() == null) {
+        throw new AxelorException(
+            0,
+            I18n.get(StockExceptionMessage.STORED_QUANTITY_IS_ZERO)
+                + " "
+                + storedProducts.getStoredProduct().getFullName());
+      }
+      if (storedProducts.getStockMoveLine() != null && storedProducts.getTrackingNumber() != null) {
+        throw new AxelorException(
+            0,
+            I18n.get(StockExceptionMessage.ALREADY_STORED_PRODUCT)
+                + " "
+                + storedProducts.getStoredProduct()
+                + " "
+                + storedProducts.getTrackingNumber().getTrackingNumberSeq());
+      } else if (storedProducts.getStockMoveLine() != null
+          && storedProducts.getTrackingNumber() == null) {
+        throw new AxelorException(
+            0,
+            I18n.get(StockExceptionMessage.ALREADY_STORED_PRODUCT)
+                + " "
+                + storedProducts.getStoredProduct());
+      }
+    }
+
+    if (checkIfAllStoredProductAreStored(storedProducts.getMassStockMove())) {
+      storedProducts.getMassStockMove().setStatusSelect(3);
```

Comment:
Please, declare constants for status to use them with StoredProductRepository.STATUS